system-addon/lib/TopStoriesFeed.jsm

```diff
@@ -311,7 +311,7 @@ this.TopStoriesFeed = class TopStoriesFeed {
         // Create a new array with a spoc inserted at index 2
         const position = SectionsManager.sections.get(SECTION_ID).order;
         let rows = this.store.getState().Sections[position].rows.slice(0, this.stories.length);
-        rows.splice(2, 0, spocs[0]);
+        rows.splice(2, 0, Object.assign(spocs[0], {pinned: true}));
```

Comment:
This modifies the `spoc[0]` object making future references also `pinned`, which I guess is fine. Would it make more sense to just treat all as pinned when it first gets assigned?

Reply:
We could do it here for all spocs: https://github.com/mozilla/activity-stream/blob/master/system-addon/lib/TopStoriesFeed.jsm#L97

But, I guess, only a subset of spocs in the feed will ever have a chance of being displayed for a given client. If you prefer, I can move it up and mark them all as pinned.

Reply:
We've typically avoided modifying objects with `Object.assign` (and mostly to create a new object) as they're harder to track down, but this should be okay.